File: salon_api/bookings.php
<?php
header("Content-Type: application/json");
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Methods: GET, POST");

$conn = new mysqli("localhost", "root", "", "salon_app");

if ($conn->connect_error) {
    die(json_encode(["success" => false, "message" => "Koneksi gagal"]));
}

$method = $_SERVER['REQUEST_METHOD'];
$action = $_GET['action'] ?? '';

// ✅ GET COUNT - Hitung jumlah booking user
if ($method == 'GET' && $action == 'count') {
    $user_id = $_GET['user_id'] ?? 0;
    
    $result = $conn->query("SELECT COUNT(*) as total FROM bookings WHERE user_id = $user_id");
    $row = $result->fetch_assoc();
    
    echo json_encode([
        "success" => true,
        "count" => $row['total']
    ]);
}

// ✅ GET - Ambil semua booking user
elseif ($method == 'GET') {
    $user_id = $_GET['user_id'] ?? 0;
    
    $result = $conn->query("
        SELECT * FROM bookings 
        WHERE user_id = $user_id 
        ORDER BY created_at ASC
    ");
    
    $bookings = [];
    while ($booking = $result->fetch_assoc()) {
        // Ambil items untuk booking ini
        $items_result = $conn->query("
            SELECT bi.*, s.name, s.image, s.category, s.duration, s.description, 
                   s.rating, s.reviews, s.is_best_seller
            FROM booking_items bi
            JOIN services s ON bi.service_id = s.id
            WHERE bi.booking_id = '{$booking['id']}'
        ");
        
        $items = [];
        while ($item = $items_result->fetch_assoc()) {
            $items[] = [
                "service" => [
                    "id" => $item["service_id"],
                    "name" => $item["name"],
                    "price" => $item["price"],
                    "rating" => $item["rating"],
                    "reviews" => $item["reviews"],
                    "category" => $item["category"],
                    "image" => $item["image"],
                    "description" => $item["description"],
                    "duration" => $item["duration"],
                    "is_best_seller" => $item["is_best_seller"] == 1 ? "1" : "0"
                ],
                "quantity" => $item["quantity"]
            ];
        }
        
        $bookings[] = [
            "id" => $booking["id"],
            "customer_name" => $booking["customer_name"],
            "booking_date" => $booking["booking_date"],
            "time_slot" => $booking["time_slot"],
            "total_price" => $booking["total_price"],
            "payment_method" => $booking["payment_method"],
            "status" => $booking["status"],
            "created_at" => $booking["created_at"],
            "items" => $items
        ];
    }
    
    echo json_encode([
        "success" => true,
        "data" => $bookings
    ]);
}

// ✅ POST - Buat booking baru
elseif ($method == 'POST') {
    $user_id = $_POST['user_id'] ?? 0;
    $booking_id = $_POST['booking_id'] ?? '';
    $customer_name = $_POST['customer_name'] ?? '';
    $booking_date = $_POST['booking_date'] ?? '';
    $time_slot = $_POST['time_slot'] ?? '';
    $total_price = $_POST['total_price'] ?? 0;
    $payment_method = $_POST['payment_method'] ?? '';
    $status = $_POST['status'] ?? 'pending';
    $items = json_decode($_POST['items'], true);
    
    // Insert booking
    $stmt = $conn->prepare("
        INSERT INTO bookings (id, user_id, customer_name, booking_date, time_slot, total_price, payment_method, status) 
        VALUES (?, ?, ?, ?, ?, ?, ?, ?)
    ");
    $stmt->bind_param("sissssss", $booking_id, $user_id, $customer_name, $booking_date, $time_slot, $total_price, $payment_method, $status);
    
    if ($stmt->execute()) {
        // Insert booking items
        foreach ($items as $item) {
            $service_id = $item['service_id'];
            $quantity = $item['quantity'];
            $price = $item['price'];
            
            $conn->query("
                INSERT INTO booking_items (booking_id, service_id, quantity, price) 
                VALUES ('$booking_id', $service_id, $quantity, $price)
            ");
        }
        
        echo json_encode([
            "success" => true,
            "message" => "Booking berhasil dibuat",
            "booking_id" => $booking_id
        ]);
    } else {
        echo json_encode([
            "success" => false,
            "message" => "Gagal membuat booking: " . $stmt->error
        ]);
    }
}

$conn->close();
?>

File: project_salon/lib/services/bookings.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/booking_model.dart';
import '../models/service_model.dart';

class BookingService {
  static const String baseUrl = "http://10.0.2.2/uas-pbm/salon_api";

  // ✅ TAMBAH - Hitung jumlah booking
  static Future<int> getBookingCount(int userId) async {
    try {
      final response = await http.get(
        Uri.parse('$baseUrl/bookings.php?action=count&user_id=$userId'),
      );

      if (response.statusCode == 200) {
        final data = jsonDecode(response.body);
        if (data['success'] == true) {
          return int.parse(data['count'].toString());
        }
      }
      return 0;
    } catch (e) {
      print("❌ Error getBookingCount: $e");
      return 0;
    }
  }
  
  // ✅ Ambil semua booking user
  static Future<List<Booking>> getBookings(int userId) async {
    try {
      final response = await http.get(
        Uri.parse('$baseUrl/bookings.php?user_id=$userId'),
      );

      print("📡 Response bookings: ${response.body}");

      if (response.statusCode == 200) {
        final data = jsonDecode(response.body);
        
        if (data['success'] == true) {
          final List bookingList = data['data'];
          return bookingList.map((bookingData) {
            return Booking(
              id: bookingData['id'],
              customerName: bookingData['customer_name'],
              bookingDate: DateTime.parse(bookingData['booking_date']),
              timeSlot: bookingData['time_slot'],
              totalPrice: int.parse(bookingData['total_price'].toString()),
              paymentMethod: bookingData['payment_method'],
              status: bookingData['status'],
              createdAt: DateTime.parse(bookingData['created_at']),
              items: (bookingData['items'] as List).map((item) {
                final serviceData = item['service'];
                return BookingItem(
                  service: Service(
                    id: int.parse(serviceData['id'].toString()),
                    name: serviceData['name'],
                    price: int.parse(serviceData['price'].toString()),
                    rating: double.parse(serviceData['rating'].toString()),
                    reviews: int.parse(serviceData['reviews'].toString()),
                    category: serviceData['category'],
                    image: serviceData['image'],
                    description: serviceData['description'],
                    duration: serviceData['duration'],
                    isBestSeller: serviceData['is_best_seller'] == "1",
                  ),
                  quantity: int.parse(item['quantity'].toString()),
                );
              }).toList(),
            );
          }).toList();
        }
      }
      return [];
    } catch (e) {
      print("❌ Error getBookings: $e");
      return [];
    }
  }

  // ✅ Buat booking baru
  static Future<bool> createBooking({
    required int userId,
    required Booking booking,
  }) async {
    try {
      // Convert items ke JSON
      final itemsJson = booking.items.map((item) => {
        'service_id': item.service.id,
        'quantity': item.quantity,
        'price': item.service.price,
      }).toList();

      final response = await http.post(
        Uri.parse('$baseUrl/bookings.php'),
        body: {
          'user_id': userId.toString(),
          'booking_id': booking.id,
          'customer_name': booking.customerName,
          'booking_date': '${booking.bookingDate.year}-${booking.bookingDate.month.toString().padLeft(2, '0')}-${booking.bookingDate.day.toString().padLeft(2, '0')}',
          'time_slot': booking.timeSlot,
          'total_price': booking.totalPrice.toString(),
          'payment_method': booking.paymentMethod,
          'status': booking.status,
          'items': jsonEncode(itemsJson),
        },
      );

      print("📡 Response create booking: ${response.body}");

      final data = jsonDecode(response.body);
      return data['success'] == true;
    } catch (e) {
      print("❌ Error createBooking: $e");
      return false;
    }
  }
}